fetch.bs

```diff
@@ -8769,82 +8769,149 @@ relevant to you. The rest of the parameters are used less frequently, often for
- <dd><p>Takes an algorithm that will be passed nothing. Indicates <a for=/>request</a>'s
- <a for=request>body</a> has been transmitted. Most standards will not need this.
+<p>Once the <a for=/>request</a> is set up, to determine which algorithms to pass to
+<a for=/>fetch</a>, determine how you would like to process the <a for=/>response</a>, and in
+particular at what stage you would like to receive a callback:
 
- <dt><a for=fetch><i>processEarlyHintsResponse</i></a>
- <dd><p>Takes an algorithm that will be passed a <a for=/>response</a> (whose
- <a for=response>status</a> is 103). Can only be used for navigations as defined by
- <cite>HTML</cite>. [[HTML]]
-
- <dt><a for=fetch><i>processResponse</i></a>
+<dl>
+ <dt>Upon completion
  <dd>
-  <p>Takes an algorithm that will be passed a <a for=/>response</a>. Indicates
-  <a for=/>response</a>'s <a for=response>header list</a> has been received and initialized. This
-  is primarily useful for standards that want to operate on <a for=/>response</a>'s
-  <a for=response>body</a>'s <a for=body>stream</a> directly.
-
-  <p>If the <a for=/>request</a>'s <a for=request>mode</a> is "<code>navigate</code>" and its
-  <a for=request>redirect mode</a> is "<code>manual</code>", then callers need to follow a very
-  specific flow with this algorithm to get the intended behavior. They should compute the
-  appropriate <a for=response>location URL</a>, and if it is non-null or failure, then they should
-  call <a for="fetch controller">process the next manual redirect</a>. This will result in
-  <a for=fetch><i>processResponse</i></a> being called again, with the next <a for=/>response</a>
-  in the redirect chain.
-
- <dt><a for=fetch><i>processResponseEndOfBody</i></a>
- <dd><p>Takes an algorithm that will be passed a <a for=/>response</a>. Indicates the network is
- done transmitting the response. This does not read <a for=/>response</a>'s
- <a for=response>body</a>.
+  <p>This is the most common way in which invokers handle a <a for=/>response</a>, for example
+  <a lt="fetch a classic script">scripts</a> and <a lt="fetch a style resource">style resources</a>.
+  The <a for=/>response</a>'s <a for=response>body</a> is obtained in its entirety into a
+  <a>byte sequence</a>, and then processed by the client.
+
+  <p>To process a <a for=/>response</a> upon completion, pass an algorithm as the
+  <a for=fetch><i>processResponseConsumeBody</i></a> argument of <a for=/>fetch</a>. The given
+  algorithm is passed a <a for=/>response</a> and null, failure, or a <a>byte sequence</a>.
+  The second argument would be null if, for example, the <a for=/>response</a> is a
+  <a>network error</a>, has a <a>null body status</a> (e.g. 204), or if the <a for=/>request</a>'s
+  <a for=request>method</a> is `<code>HEAD</code>. It would be failure if there was an I/O error
+  while reading the <a for=response>body</a>, or a <a>byte sequence</a> representing the
+  successfully obtained <a for=response>body</a>.
```

Comment:
```suggestion
  while reading the <a for=response>body</a>. It would be a <a>byte sequence</a> representing the
  <a for=response>body</a> if the body is successfully read.
```